dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -131,6 +137,34 @@ LRESULT CSearchBar::OnSearchButtonClicked(WORD wNotifyCode, WORD wID, HWND hWndC
         return 0;
     }
 
+    // See if we have a szFileName, our first character is not an asterisk,
+    // and we have room in MAX_PATH to add an asterisk at the beginning of szFileName then add it
+    if(wcscmp(pSearchStart->szFileName, L"") != 0)
+        {
+        len = wcslen(pSearchStart->szFileName);
+        if(len > 0)
+            {
+            ptrchar = pSearchStart->szFileName[len - 1];
+            memcpy(endchar, &ptrchar, 2);
+            ptrchar = pSearchStart->szFileName[0];
+            memcpy(startchar, &ptrchar, 2);
+            if ((len < MAX_PATH) && (wcscmp(startchar, L"*") != 0))
+                {
+                    memmove(&pSearchStart->szFileName[1], &pSearchStart->szFileName[0],
+                            len * sizeof(WCHAR) + sizeof(WCHAR));
+                    len = len + 1;
+                    pSearchStart->szFileName[0] = asterisk[0];
+                }
+            }
+
+        // See if our last character is an asterisk and if not and we have room then add one
+        if((wcscmp(endchar, L"*") != 0) && (len < MAX_PATH))
```

Comment:
`if<space>(...)`

Reply:
`if ((len < MAX_PATH) && (endchar == L'*'))`

Reply:
When I use this syntax under RosBE 2.1.6 running on Win 7 I receive the following error:

../dll/win32/browseui/shellfind/CSearchBar.cpp: In member function 'LRESULT CSea
rchBar::OnSearchButtonClicked(WORD, WORD, HWND, BOOL&)':
../dll/win32/browseui/shellfind/CSearchBar.cpp:161:45: error: ISO C++ forbids co
mparison between pointer and integer [-fpermissive]
At global scope:
cc1plus.exe: error: unrecognized command line option "-Wno-unused-const-variable
" [-Werror]
cc1plus.exe: all warnings being treated as errors
ninja: build stopped: subcommand failed.

Do you have a suggestion on how to fix this? Thanks.

Reply:
The suggestion works only if endchar is declared as `WCHAR endchar;` ; see my other comments.

Reply:
OK, I added a new "WCHAR endchar1:" and then changed to "endchar1 = *ptrchar;" and now this:
../dll/win32/browseui/shellfind/CSearchBar.cpp: In member function 'LRESULT CSea
rchBar::OnSearchButtonClicked(WORD, WORD, HWND, BOOL&)':
../dll/win32/browseui/shellfind/CSearchBar.cpp:150:25: error: invalid type argum
ent of unary '*' (have 'WCHAR {aka wchar_t}')
At global scope:
Suggestions now?